### PR TITLE
Add ref/out keywords on arguments for shadow methods in GeneratedComInterface

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Fix a regression blocking https://github.com/dotnet/sdk/pull/43015

### CODE REVIEW

**Review by jtschuster — APPROVED** [review]
nit: I don't think the .sln changes are necessary.

**Comment by AaronRobinsonMSFT** [comment]
Is this a regression? If so, what did we miss?

File: src/libraries/System.Runtime.InteropServices/tests/ComInterfaceGenerator.Unit.Tests/CodeSnippets.cs [hunk 1]
```diff
@@ -671,6 +671,36 @@ interface IOtherInterface
...
+                [PreserveSig]
+                int FindValue(int key, out IValue? value);
+            }
+
+            {{GeneratedComInterface()}}
+            unsafe partial interface INativeDerived : INativeAPIBase
+            {
+                [PreserveSig]
+                int GetName(out char* name);
+            }
```
Comment (AaronRobinsonMSFT) [inline src/libraries/System.Runtime.InteropServices/tests/ComInterfaceGenerator.Unit.Tests/CodeSnippets.cs:701]: Do we already have a test for `ref`? What about `in`?

Comment (AaronRobinsonMSFT) [reply src/libraries/System.Runtime.InteropServices/tests/ComInterfaceGenerator.Unit.Tests/CodeSnippets.cs:701]: If not, feel free to add them to the existing signature. or a new one for kind.

Also, `ref readonly`?

Comment (jkoritzinsky) [reply src/libraries/System.Runtime.InteropServices/tests/ComInterfaceGenerator.Unit.Tests/CodeSnippets.cs:701]: I'll refactor this test to validate all of the kinds.

**Comment by jkoritzinsky** [comment]
Yeah this was a regression from one of the refactorings I did when I simplified the bound generator concept in the source generator APIs (and moved this part to not use that infrastructure)

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Review by elinor-fung — APPROVED** [review]

**Comment by jkoritzinsky** [comment]
Looks like we have one more bug in here. I'll push out a fix.

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Pull request merged**